packages/wonder-blocks-dropdown/components/action-menu.js

```diff
@@ -0,0 +1,237 @@
+    static defaultProps = {
+        alignment: "left",
+        containsSelectionItems: false,
+        disabled: false,
+    };
+
+    constructor(props: MenuProps) {
+        super(props);
+
+        this.state = {
+            open: false,
+        };
+    }
+
+    toggleMenu() {
+        this.setState((prevState) => ({
+            open: !prevState.open,
+        }));
+    }
+
+    handleSelected(selectedValue: string, newSelectionState: boolean) {
+        const {onChange, selectedValues} = this.props;
+
+        // If either of these are not defined, return.
+        if (!onChange || !selectedValues) {
+            return;
+        }
+
+        if (newSelectionState) {
+            onChange([...selectedValues, selectedValue].sort());
+        } else {
+            const updatedSelection = selectedValues;
+            updatedSelection.splice(selectedValues.indexOf(selectedValue), 1);
+            onChange(updatedSelection);
+        }
+    }
+
+    render() {
+        const {
+            alignment,
+            containsSelectionItems,
+            disabled,
+            items,
+            menuText,
+            selectedValues,
+            style,
+        } = this.props;
+
+        const {open} = this.state;
+
+        const opener = (
+            <ActionMenuOpener
+                disabled={disabled}
+                onClick={() => this.toggleMenu()}
+                style={style}
+            >
+                {menuText}
+            </ActionMenuOpener>
+        );
+
+        const menuItems = items.map((item, index) => {
+            if (item.type === "separator") {
+                return <SeparatorItem key={index} />;
+            } else if (item.type === "select") {
+                return (
+                    <SelectItem
+                        key={index}
+                        disabled={item.disabled}
+                        label={item.label}
+                        onClick={item.onClick}
+                        onToggle={(value, state) =>
+                            this.handleSelected(value, state)
+                        }
+                        selected={
+                            selectedValues
+                                ? selectedValues.includes(item.value)
+                                : false
+                        }
+                        value={item.value}
+                        variant={"check"}
+                    />
+                );
+            } else if (item.type === "action") {
+                return (
+                    <ActionItem
+                        key={index}
+                        disabled={item.disabled}
+                        indent={containsSelectionItems}
+                        label={item.label}
+                        href={item.href}
+                        clientNav={item.clientNav}
+                        onClick={item.onClick}
+                    />
+                );
+            }
```

Comment:
It seems find to me to not have an `else` clause since `flow` keeps us safe.

Reply:
I'll leave this in because I think it's slightly easier to understand for someone new looking at the clause.